F-Droid badge & app links

endingisnight opened the pull request:
#831 
- added the F-Droid badge (with many localizastions)
- added Shitter, Husky and Raccoon
- made horrible looking links to the app's pages (WHY IS THE ICON SMALLER?!)
  ![image](https://github.com/user-attachments/assets/ef211e28-90b1-4c55-81ac-2ac46868375b)
- learned some react :)

andypiper commented:
As an FYI, I am on vacation for a week or so and may not get to this immediately. I do not intend to add two of these apps (Shitter based on the name which does not align with my goal to keep the project site somewhat professional, and Raccoon based on potential confusion with “for Friendica” being in the title, even if it does work with Mastodon) - per the [adding apps](https://github.com/mastodon/joinmastodon/blob/main/CONTRIBUTING.md#adding-apps) section of the contributing guide, we reserve the right not to include every app that exists.

A quick glance at the work you’ve done to add the F-Droid badge looks good but I’ll do a proper review when I’m back and may request changes then. Thanks!

endingisnight commented:
I get not including Shitter (unless they change their name https://github.com/nuclearfog/Shitter/issues/59), but with "Raccoon for Friendica" I'll say that while installed, it shows up as just "Raccoon". So would it be fine to just exclude the "for Friendica" part and still include it?

andypiper commented:
Yes, I think that would be an OK approach.

endingisnight commented:
removed "tooot" since it's been [archived](https://github.com/tooot-app/app)

endingisnight commented:
Alright, the creator of Shitter, nuclearfog has moved to [codeberg](https://mastodon.social/@nuclearfog/113963620594407097) (very cool), changed the name from [Shitter to Smither](https://mastodon.social/@nuclearfog/113975439826437032) and has [submitted](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/19483) the changes to F-Droid. :)

endingisnight commented:
Okay so hear me out, I got a bit carried away and added more categories AND more data to the apps including app store sources' (https://github.com/mastodon/joinmastodon/issues/326) and links to the source code (https://github.com/mastodon/joinmastodon/issues/369). I also removed my previous icons that showed next to the cards since there was no way that all the store source icons were gonna fit. I came across https://github.com/mastodon/joinmastodon/issues/704 as well which (after pulling my hair our) I fixed.

endingisnight commented:
https://github.com/mastodon/joinmastodon/issues/324

![:O](https://github.com/user-attachments/assets/f6bcdf68-43a4-4f91-bd47-53d25ad53f6b)

Poussinou commented:
It looks like a lot of work and a good PR.

I'm not sure if IzzyOnDroid repo should be included, this might be confusing for most of the users (adding a repo inside F-Droid is not something everybody know how to do).

There is also a Android mobile app developed by KDE, [Tokodon](https://invent.kde.org/network/tokodon).

endingisnight commented:
@Poussinou Izzy also has it's own (unofficial) app which has the repo pre added, but oh well.
Also Tokodon is also for ANDROID? Not just plasma mobile? The source dosent link to fdroid or gplay, or am I blind?

ChaosExAnima commented:
Hi @endingisnight! Thank you for your contribution. We agree that we should have more platforms displaying on this page, however we're trying to not make the UI too complex for people trying to join Mastodon. I discussed it with the team and we think this is a good problem for our (currently being recruited) designer to tackle.

Sorry for the long wait- I'm going through the repo to resolve all open issues, and because of that I'm going to close this one. However I appreciate your work here and I hope this doesn't discourage future contributions! 🙂

endingisnight commented:
@ChaosExAnima "we're trying to not make the UI too complex" mhhh, I do wonder if adding another dropdown for filtering for a specific store would make the UI less complex? So the user would pick a platform (android/iOS) (maybe even default to based on UA?) and then a store (fdroid/playstore/sidestore/etc).

Also, should I make a separate PR for just the fdroid badge?

ChaosExAnima commented:
> "we're trying to not make the UI too complex" mhhh, I do wonder if adding another dropdown for filtering for a specific store would make the UI less complex? So the user would pick a platform (android/iOS) (maybe even default based on UA?) and then a store (fdroid/playstore/sidestore/etc).

idk, that's a question for our future UX designer. i think there's a few ways we can improve this, but until we have someone we're going to hold off on changes to this.

> Also, should I make a separate PR for just the fdroid badge?

the only place we'd want to put that badge would be to [link to the official Mastodon package,](https://f-droid.org/en/packages/org.joinmastodon.android/) unless you were thinking about somewhere else? i don't see a problem with adding that next to the official Google app store badge- thoughts @andypiper?

regardless: please do not create SVGs for every language. a simple SVG that just says "F-Droid" is best.
